app/controllers/short_url_requests_controller.rb

```diff
@@ -24,12 +24,16 @@ def create
     @short_url_request.requester = current_user
     @short_url_request.contact_email = current_user.email
 
-    if @short_url_request.save
-      Notifier.short_url_requested(@short_url_request).deliver_now
-      flash[:success] = "Your request has been made."
-      redirect_to root_path
+    if @short_url_request.duplicate? && !@short_url_request.confirmed
```

Comment:
We could move these checks into a single method on the model perhaps: `@short_url_request.requires_confirmation` ?